sql/core/src/main/scala/org/apache/spark/sql/execution/adaptive/AQEPropagateEmptyRelation.scala

```diff
@@ -34,11 +35,14 @@ import org.apache.spark.sql.execution.joins.HashedRelationWithAllNullKeys
  */
 object AQEPropagateEmptyRelation extends PropagateEmptyRelationBase {
   override protected def isEmpty(plan: LogicalPlan): Boolean =
-    super.isEmpty(plan) || (!isRootRepartition(plan) && getEstimatedRowCount(plan).contains(0))
+    super.isEmpty(plan) || plan.isInstanceOf[EmptyRelation] ||
```

Comment:
do we need to check the type explicitly? It has row count which should be sufficient.

Reply:
ping @liuzqt , I think this change is not needed as we have https://github.com/apache/spark/pull/46830/files#diff-d72cf9d12043beb25760fb2755ab0dd7fdd881483068c833541a126a1a38b780R32

Reply:
@cloud-fan  we still need to check this explicitly. `getEstimatedRowCount` only match for specific patterns. Before the change, the reason why it works is because `super.isEmpty` explicitly match empty `LocalRelation`.


We can also match `EmptyRelation` in `getEstimatedRowCount` alternatively. I don't have perfernce.

Reply:
I think it's more reasonable to make `getEstimatedRowCount` recognize this new `EmptyRelation`